microsoft/testsuites/kselftest/kselftest.py

```diff
@@ -64,66 +65,89 @@ def __init__(
             self._remote_tar_path = self.get_tool_path(
                 use_global=True
             ) / os.path.basename(self._tar_file_path)
-        else:
-            self._remote_tar_path = (
-                self.get_tool_path(use_global=True)
-                / f"{self._KERNEL_REPO_NAME}{self._KSELFTEST_TAR_PATH}"
-            )
 
         # command to run kselftests
-        if not self._tar_file_path:
-            self._command = self.get_tool_path(use_global=True) / "run_kselftest.sh"
-        else:
-            self._command = (
-                self.get_tool_path(use_global=True)
-                / "kselftest-packages/run_kselftest.sh"
-            )
+        self._kself_installed_dir = (
+            self.get_tool_path(use_global=True) / "kselftest-packages"
+        )
+
+        self._command = self._kself_installed_dir / "run_kselftest.sh"
 
     # install common dependencies
     def _install(self) -> bool:
-        if not isinstance(self.node.os, CBLMariner):
+        if not (
+            isinstance(self.node.os, Ubuntu) or isinstance(self.node.os, CBLMariner)
+        ):
             raise UnsupportedDistroException(
-                self.node.os, "kselftests are supported on Mariner VMs only."
+                self.node.os, "kselftests are supported on Ubuntu and Mariner only"
```

Comment:
The error message of `UnsupportedDistroException` has OS information already, it doesn't need to mention here. The message should be like `kselftests in LISA doesn't support this os`.

Reply:
Thanks, Updated